corehq/apps/users/views/mobile/users.py

```diff
@@ -1301,47 +1321,86 @@ def _excel_data(self, rows):
+    return _count_users(request, domain, MOBILE_USER_TYPE)
+
+
+@require_can_edit_web_users
+@require_can_use_filtered_user_download
+def count_web_users(request, domain):
+    return _count_users(request, domain, WEB_USER_TYPE)
+
+
+@login_and_domain_required
+def _count_users(request, domain, user_type):
+    if user_type not in [MOBILE_USER_TYPE, WEB_USER_TYPE]:
+        raise AssertionError(f"Invalid user type for _count_users: {user_type}")
+
+    from corehq.apps.users.dbaccessors import (
+        count_mobile_users_by_filters,
+        count_web_users_by_filters,
+        count_invitations_by_filters,
+    )
+    form = UserFilterForm(request.GET, domain=domain, couch_user=request.couch_user, user_type=user_type)
     if form.is_valid():
         user_filters = form.cleaned_data
     else:
         return HttpResponseBadRequest("Invalid Request")
+
     user_count = 0
-    for domain in user_filters['domains']:
-        user_count += get_commcare_users_by_filters(domain, user_filters, count_only=True)
+    (is_multi_domain_download, domains_list) = get_domains_from_user_filters(domain, user_filters)
+    for current_domain in domains_list:
+        if user_type == MOBILE_USER_TYPE:
+            user_count += count_mobile_users_by_filters(current_domain, user_filters)
+        else:
+            user_count += count_web_users_by_filters(current_domain, user_filters)
+            user_count += count_invitations_by_filters(current_domain, user_filters)
+
     return json_response({
         'count': user_count
     })
 
 
 @require_can_edit_or_view_commcare_users
 def download_commcare_users(request, domain):
-    form = CommCareUserFilterForm(request.GET, domain=domain, couch_user=request.couch_user)
+    return download_users(request, domain, user_type=MOBILE_USER_TYPE)
+
+
+@login_and_domain_required
+def download_users(request, domain, user_type):
+    if user_type not in [MOBILE_USER_TYPE, WEB_USER_TYPE]:
+        raise AssertionError(f"Invalid user type for download_users: {user_type}")
+
+    form = UserFilterForm(request.GET, domain=domain, couch_user=request.couch_user, user_type=user_type)
     if form.is_valid():
         user_filters = form.cleaned_data
     else:
-        return HttpResponseRedirect(
-            reverse(FilteredUserDownload.urlname, args=[domain]) + "?" + request.GET.urlencode())
+        view = FilteredCommCareUserDownload if user_type == MOBILE_USER_TYPE else FilteredWebUserDownload
+        return HttpResponseRedirect(reverse(view, args=[domain]) + "?" + request.GET.urlencode())
     download = DownloadBase()
     if form.cleaned_data['domains'] != [domain]:  # if additional domains added for download
-        track_workflow(request.couch_user.username, 'Domain filter used for mobile download')
-    is_web_download = False
-    if form.cleaned_data['columns'] == CommCareUserFilterForm.USERNAMES_COLUMN_OPTION:
+        user_type = 'mobile' if user_type == MOBILE_USER_TYPE else 'web'
```

Comment:
So,

the passed user_type is already either mobile/web? Not sure why this if is needed here.

Reply:
just defensive

Reply:
I replied to this from the whole PR view, without looking at the whole context. Removed `user_type` in 82781343579